core/drivers/gic.c

```diff
@@ -267,6 +304,34 @@ static void gic_it_disable(struct gic_data *gd, size_t it)
 	write32(mask, gd->gicd_base + GICD_ICENABLER(idx));
 }
 
+static void gic_it_pending(struct gic_data *gd, size_t it)
+{
+	size_t idx = it / NUM_INTS_PER_REG;
+	uint32_t mask = 1 << (it % NUM_INTS_PER_REG);
+
+	assert(it <= gd->max_it); /* Not too large */
+	/* Assigned to group0 */
+	assert(!(read32(gd->gicd_base + GICD_IGROUPR(idx)) & mask));
+	/* not pending yet */
+	assert(!(read32(gd->gicd_base + GICD_ICPENDR(idx)) & mask));
```

Comment:
So I just remove the assert, or I should clear the pending state before setting the interrupt pending?